internal/builders/golang/targets.go

```diff
@@ -133,7 +134,11 @@ var (
 
 func goVersion(build config.Build) ([]byte, error) {
 	cmd := exec.Command(build.GoBinary, "version")
-	cmd.Dir = build.Dir // Set Dir to build directory in case of reletive path to GoBinary
+	// If the build.Dir is acessible, set the cmd dir to it in case
+	// of reletive path to GoBinary
+	if _, err := os.Stat(build.Dir); err == nil {
```

Comment:
A potentially faster way might be to check if `Dir` and `GoBinary` are filled out, and if `GoBinary` is relative. Since that is the only instance that we need to actually change directory when running GoBinary. Not 100% sure if the string operations would for sure faster than the syscall, it would need some testing.

Maybe something like:

```suggestion
	if build.Dir != "" && build.GoBinary != "go" && (!strings.HasPrefix(build.GoBinary, "/") && strings.Contains(build.GoBinary, "/")){
```

Reply:
I think its ok as is...